java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java

```diff
@@ -5,8 +5,24 @@
 package com.zeroc.IceInternal;
 
 public interface Transceiver {
+  /**
+   * Returns the selectable channel used by the thread pool's selector to wait for read or
+   * write readiness.
+   *
+   * @return The selectable channel that will be registered with the thread pool's selector
+   * or null if the transceiver doesn't use a selectable channel.
+   */
   java.nio.channels.SelectableChannel fd();
 
+  /**
+   * Sets the transceiver ready callback. This method is called by the thread pool to provide
+   * a callback object that the transceiver can use to notify the thread pool's selector when
```

Comment:
"can use" should be uses?

I assume a correct implementation of Transceiver does not ignore this ReadyCallback. If it has a callback, it calls it whenever it's ready?

Reply:
The thread pool always calls `setReadyCallback` with a non-null callback object. The transport is free to use it or not use it. For example:
- the TCP transceiver  implementation ignores it
-  the SSL and WS transceivers use it (in addition to the selectable channel)
- the BT transceiver implementation doesn't have a selectable channel, it only relies on the ready callback for read/write readiness